pgxn/neon/neon.c

```diff
@@ -522,29 +539,44 @@ RestoreRunningXactsFromClog(CheckPoint *checkpoint, TransactionId **xids, int *n
 			elog(LOG, "too many running xacts to restore from the CLOG; oldestXid=%u oldestActiveXid=%u nextXid %u",
 				 checkpoint->oldestXid, checkpoint->oldestActiveXid,
 				 XidFromFullTransactionId(checkpoint->nextXid));
-			goto fail;
+
+			overflow = true;
+			switch (running_xacts_overflow_policy)
+			{
+				case OP_WAIT:
```

Comment:
And related to the same question, since we now going to use this new GUC, I think it makes sense to start using it in regression tests. At least those that are related to physical replication
- `test_replica_start.py` (I see you used 'wait' there. Maybe add another case? Is there an easy way to reproduce such overflow?)
- `test_physical_replication.py`

Can you make them using the value you propose for prod?

Reply:
This PR and introduced GUC comes into play only when known xacts are overflows.
There is only one test now checking it: test_replica_start.py

It is no so easy to reproduce known xacts overflow: we need transaction with larger number of subtransactions.
I can be created using function catching exceptions in the loop.
But to challenge is to delay commit of this transaction until its changes are propagated to replica. And prelim is not able to start until it receives non-overflown snapshot. The problem is somehow solved in test_replica_start.py.

But I do no see that playing with this GIC in other tests will have any effect or sense. Especially if we create this GUC mostly as temporary workaround.

There is yet another way to achieve known subxacts overflow - which we actually faced in prod: restart primary many times until xid alignment produce large number of holes. With current alignment on 1024, just 4 such restart may be enough. With new @hlinnaka PR changing alignment to 128 we need 8 times more  restarts. But I do not quite understand what we are expected to check with this test? Prove that ignoring overflow can cause problems at replica? Or that replica will be able to start with `neon.unning_xacts_overflow_policy=ignore`?

In any case, I can add such test if you think that it will be useful.